openhtf/util/measurements.py

```diff
@@ -138,6 +138,19 @@ def WithValidator(self, validator):
     self.validators.append(validator)
     return self
 
+  def WithArgs(self, **kwargs):
+    """Creates a new Measurement, see openhtf.PhaseInfo.WithArgs."""
+    new_meas = mutablerecords.CopyRecord(self)
+    if '{' in new_meas.name:
+      formatter = lambda x: x.format(**kwargs)
+    else:
+      # str % {'a': 1} is harmless if str doesn't use any interpolation.
```

Comment:
I can check if it starts with a number here, but I can only check alphanumeric later since `WithArgs` may be chained. Could I get this in without that for now?